json_spirit/json_spirit_writer_template.h

```diff
@@ -25,13 +25,9 @@ namespace json_spirit
         return 'A' - 10 + ch;
     }
 
-#pragma GCC diagnostic push

```

Comment:
I know. But we added macros to suppress warning about unused typedef. The same must be done for clang. I think it is better to remove that unused typedef.